packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -214,6 +229,17 @@ export const GridColumnHeaderItem = React.memo(
           height={headerHeight}
           {...resizeEventHandlers}
         />
+        <GridColumnHeaderMenu
+          columnMenuId={columnMenuId}
+          columnMenuButtonId={columnMenuButtonId}
+          field={column.field}
+          open={columnMenuOpen}
+          target={iconButtonRef.current}
+          ContentComponent={apiRef!.current.components.ColumnMenu}
+          contentComponentProps={{
+            ...apiRef!.current.componentsProps?.columnMenu,
```

Comment:
this will create a new ref at every render? Why do we need to clone it?

Reply:
I'm not creating a ref at every render, their values are kept across renders. Could you be more specific about the clone problem? I can't see it.

Reply:
if GridColumnHeaderMenu is memoized then it will be broken because you pass a new contentComponentProps every time the GridColumnHeaderItem renders
Why do you clone?

Reply:
It was already there when I moved from GridComponent. It was added in [this](https://github.com/mui-org/material-ui-x/pull/890/files?file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-7b46969348f36dfdd1132e097c59264a6f22eb1fb68eb881aa02e999e45d1122R127) PR to merge with other props, then these other props were removed in [this commit](https://github.com/mui-org/material-ui-x/commit/169ba3e94b42f5c15764c0925064544e90a9f7d8#diff-7b46969348f36dfdd1132e097c59264a6f22eb1fb68eb881aa02e999e45d1122L132) but the clone didn't. I removed the clone now since it's not needed anymore.